--- src.old/comm.c	Wed Dec 27 18:17:55 1995
+++ src/comm.c	Wed Nov 12 20:36:44 1997
@@ -54,7 +54,8 @@
 #include <time.h>
 
 #include "merc.h"
-
+#include "imc.h"
+#include "icec.h"
 
 
 /*
@@ -158,8 +159,8 @@ int	socket		args( ( int domain, int type
 #endif
 
 #if	defined( linux )
-int	accept		args( ( int s, struct sockaddr *addr, int *addrlen ) );
-int	bind		args( ( int s, struct sockaddr *name, int namelen ) );
+/*int	accept		args( ( int s, struct sockaddr *addr, int *addrlen ) ); */
+/*int	bind		args( ( int s, struct sockaddr *name, int namelen ) ); */
 int	close		args( ( int fd ) );
 int	getpeername	args( ( int s, struct sockaddr *name, int *namelen ) );
 int	getsockname	args( ( int s, struct sockaddr *name, int *namelen ) );
@@ -412,9 +413,16 @@ int main( int argc, char **argv )
 #if defined( unix )
     control = init_socket( port );
     boot_db( );
+
+	imc_startup ("imc/"); /* start up IMC */
+    icec_init();
+    
     sprintf( log_buf, "EnvyMud is ready to rock on port %d.", port );
     log_string( log_buf );
     game_loop_unix( control );
+    
+    imc_shutdown(); /* shut down IMC */
+    
     close( control );
 #endif
 
@@ -694,6 +702,9 @@ void game_loop_unix( int control )
 	    FD_SET( d->descriptor, &exc_set );
 	}
 
+	/* IMC */
+	maxdesc=imc_fill_fdsets(maxdesc, &in_set, &out_set, &exc_set);
+
 	if ( select( maxdesc+1, &in_set, &out_set, &exc_set, &null_time ) < 0 )
 	{
 	    perror( "Game_loop: select: poll" );
@@ -770,6 +781,9 @@ void game_loop_unix( int control )
 	    }
 	}
 
+
+	/* IMC */
+	imc_idle_select(&in_set, &out_set, &exc_set, current_time);
 
 
 	/*
