--- src.old/merc.h	Wed Dec 27 18:17:56 1995
+++ src/merc.h	Wed Nov 12 20:36:55 1997
@@ -926,6 +926,13 @@ struct	pc_data
     int                 pagelen;
     int 		learned		[ MAX_SKILL ];
     bool                switched;
+
+    long        imc_deaf;    /* IMC channel deaf flags */
+    long        imc_allow;   /* IMC channel allow flags */
+    long        imc_deny;    /* IMC channel deny flags */
+    char *      rreply;      /* IMC reply-to */
+	char *      rreply_name; /* IMC reply-to shown to char */
+	char *ice_listen;
 };
 
 /*
