--- src.old/save.c	Wed Dec 27 18:17:55 1995
+++ src/save.c	Wed Nov 12 20:38:20 1997
@@ -166,6 +166,10 @@ void fwrite_char( CHAR_DATA *ch, FILE *f
     fprintf( fp, "Armr        %d\n",	ch->armor		);
     fprintf( fp, "Wimp        %d\n",	ch->wimpy		);
     fprintf( fp, "Deaf        %d\n",	ch->deaf		);
+    fprintf( fp, "IMC         %ld\n",   ch->pcdata->imc_deaf    );
+    fprintf( fp, "IMCAllow    %ld\n",   ch->pcdata->imc_allow   );
+    fprintf( fp, "IMCDeny     %ld\n",   ch->pcdata->imc_deny    );
+    fprintf( fp, "ICEListen   %s~\n",   ch->pcdata->ice_listen  );
 
     if ( IS_NPC( ch ) )
     {
@@ -391,6 +395,8 @@ bool load_char_obj( DESCRIPTOR_DATA *d, 
 
     ch->pcdata->switched                = FALSE;
 
+    ch->pcdata->ice_listen = str_dup("");
+	
     found = FALSE;
     fclose( fpReserve );
 
@@ -604,6 +610,10 @@ void fread_char( CHAR_DATA *ch, FILE *fp
 
 	case 'I':
 	    KEY ( "Immskll",    ch->pcdata->immskll,    fread_string( fp ) );
+	    KEY ( "IMC",        ch->pcdata->imc_deaf,   fread_number( fp ) );
+	    KEY ( "IMCAllow",   ch->pcdata->imc_allow,  fread_number( fp ) );
+		KEY ( "IMCDeny",    ch->pcdata->imc_deny,   fread_number( fp ) );
+		KEY ( "ICEListen",  ch->pcdata->ice_listen, fread_string( fp ) );
 
 	case 'L':
 	    KEY( "Lvl", 	ch->level,		fread_number( fp ) );
