*** osrc/comm.c	Sun Apr 14 19:28:46 1996
--- src/comm.c	Mon Jun  9 00:37:20 1997
***************
*** 33,38 ****
--- 33,40 ----
  #include "db.h"
  #include "house.h"
  
+ #include "imc.h"
+ 
  #ifdef HAVE_ARPA_TELNET_H
  #include <arpa/telnet.h>
  #else
***************
*** 237,246 ****
--- 239,252 ----
    signal_setup();
  #endif
  
+   imc_startup("imc/");
+ 
    log("Entering game loop.");
  
    game_loop(mother_desc);
  
+   imc_shutdown();
+ 
    log("Closing all sockets.");
    while (descriptor_list)
      close_socket(descriptor_list);
***************
*** 468,475 ****
      if (descriptor_list == NULL) {
        log("No connections.  Going to sleep.");
        FD_ZERO(&input_set);
        FD_SET(mother_desc, &input_set);
!       if (select(mother_desc + 1, &input_set, (fd_set *) 0, (fd_set *) 0, NULL) < 0) {
  	if (errno == EINTR)
  	  log("Waking up to process signal.");
  	else
--- 474,486 ----
      if (descriptor_list == NULL) {
        log("No connections.  Going to sleep.");
        FD_ZERO(&input_set);
+       FD_ZERO(&output_set);
+       FD_ZERO(&exc_set);
        FD_SET(mother_desc, &input_set);
!       
!       maxdesc = imc_fill_fdsets(mother_desc, &input_set, &output_set, &exc_set);
!       
!       if (select(maxdesc + 1, &input_set, &output_set, &exc_set, NULL) < 0) {
  	if (errno == EINTR)
  	  log("Waking up to process signal.");
  	else
***************
*** 495,500 ****
--- 506,513 ----
        FD_SET(d->descriptor, &exc_set);
      }
  
+     maxdesc = imc_fill_fdsets(maxdesc, &input_set, &output_set, &exc_set);
+ 
      /*
       * At this point, we have completed all input, output and heartbeat
       * activity from the previous iteration, so we have to put ourselves
***************
*** 647,652 ****
--- 660,667 ----
        log("Warning: Missed more than 30 seconds worth of pulses");
        missed_pulses = 30 * PASSES_PER_SEC;
      }
+ 
+     imc_idle_select(&input_set, &output_set, &exc_set, last_time.tv_sec);
  
      /* Now execute the heartbeat functions */
      while (missed_pulses--)
