*** osrc/interpreter.c	Sat Apr 13 04:39:21 1996
--- src/interpreter.c	Mon Jun  9 00:57:33 1997
***************
*** 10,15 ****
--- 10,17 ----
  
  #define __INTERPRETER_C__
  
+ #define CIRCLE
+ 
  #include "conf.h"
  #include "sysdep.h"
  
***************
*** 24,29 ****
--- 26,32 ----
  #include "mail.h"
  #include "screen.h"
  
+ #include "imc-mercbase.h"
  
  extern const struct title_type titles[NUM_CLASSES][LVL_IMPL + 1];
  extern char *motd;
***************
*** 419,424 ****
--- 422,450 ----
    { "roll"     , POS_RESTING , do_action   , 0, 0 },
    { "roomflags", POS_DEAD    , do_gen_tog  , LVL_IMMORT, SCMD_ROOMFLAGS },
    { "ruffle"   , POS_STANDING, do_action   , 0, 0 },
+ 
+   { "rchat"    , POS_SLEEPING, do_rchat    , 1, 0 },
+   { "rtell"    , POS_SLEEPING, do_rtell    , 1, 0 },
+   { "rreply"   , POS_SLEEPING, do_rreply   , 1, 0 },
+   { "rwho"     , POS_SLEEPING, do_rwho     , 1, 0 },
+   { "rquery"   , POS_SLEEPING, do_rquery   , 1, 0 },
+   { "rwhois"   , POS_SLEEPING, do_rwhois   , 1, 0 },
+   { "rfinger"  , POS_SLEEPING, do_rfinger  , 1, 0 },
+   { "rbeep"    , POS_SLEEPING, do_rbeep    , 1, 0 },
+   { "istats"   , POS_DEAD    , do_istats   , 1, 0 },
+   { "rchannels", POS_DEAD    , do_rchannels, 1, 0 },
+ 
+   { "imc"      , POS_DEAD    , do_imc      , LVL_GRGOD, 0 },
+   { "imclist"  , POS_DEAD    , do_imclist  , 1, 0 },
+   { "rcode"    , POS_SLEEPING, do_rcode    , LVL_GOD, 0 },
+   { "rimm"     , POS_DEAD    , do_rimm     , LVL_IMMORT, 0 },
+   { "rinfo"    , POS_DEAD    , do_rinfo    , LVL_IMMORT, 0 },
+   { "rsockets" , POS_DEAD    , do_rsockets , LVL_GRGOD, 0 },
+   { "rconnect" , POS_DEAD    , do_rconnect , LVL_GRGOD, 0 },
+   { "rdisconnect", POS_DEAD  , do_rdisconnect, LVL_GRGOD, 0 },
+   { "rignore"  , POS_DEAD    , do_rignore  , LVL_GRGOD, 0 },
+   { "rchanset" , POS_DEAD    , do_rchanset , LVL_GRGOD, 0 },
+   { "mailqueue", POS_DEAD    , do_mailqueue, LVL_GOD, 0 },
  
    { "say"      , POS_RESTING , do_say      , 0, 0 },
    { "'"        , POS_RESTING , do_say      , 0, 0 },
