*** osrc/mail.c	Sat Apr 13 04:39:21 1996
--- src/mail.c	Sun Jun  8 23:57:09 1997
***************
*** 26,31 ****
--- 26,33 ----
  #include "handler.h"
  #include "mail.h"
  
+ #include "imc.h"
+ 
  void postmaster_send_mail(struct char_data * ch, struct char_data *mailman,
  			  int cmd, char *arg);
  void postmaster_check_mail(struct char_data * ch, struct char_data *mailman,
***************
*** 371,377 ****
  	  "Date: %s\r\n"
  	  "  To: %s\r\n"
  	  "From: %s\r\n\r\n", tmstr, get_name_by_id(recipient),
! 	  get_name_by_id(header.header_data.from));
  
    string_size = (sizeof(char) * (strlen(buf) + strlen(header.txt) + 1));
    CREATE(message, char, string_size);
--- 373,381 ----
  	  "Date: %s\r\n"
  	  "  To: %s\r\n"
  	  "From: %s\r\n\r\n", tmstr, get_name_by_id(recipient),
! 	  get_name_by_id(header.header_data.from) ?
! 	  get_name_by_id(header.header_data.from) :
! 	  "IMC Mail System");
  
    string_size = (sizeof(char) * (strlen(buf) + strlen(header.txt) + 1));
    CREATE(message, char, string_size);
***************
*** 438,445 ****
--- 442,451 ----
  void postmaster_send_mail(struct char_data * ch, struct char_data *mailman,
  			  int cmd, char *arg)
  {
+   int imc_mail = 0;
    long recipient;
    char buf[256];
+   char *addr = NULL;
  
    if (GET_LEVEL(ch) < MIN_MAIL_LEVEL) {
      sprintf(buf, "$n tells you, 'Sorry, you have to be level %d to send mail!'",
***************
*** 460,466 ****
      act(buf, FALSE, mailman, 0, ch, TO_VICT);
      return;
    }
!   if ((recipient = get_id_by_name(buf)) < 0) {
      act("$n tells you, 'No one by that name is registered here!'",
  	FALSE, mailman, 0, ch, TO_VICT);
      return;
--- 466,480 ----
      act(buf, FALSE, mailman, 0, ch, TO_VICT);
      return;
    }
!   if (strchr(buf, '@') != NULL) {
!     imc_mail = 1;
!     addr = str_dup(buf);
!   }
!   if (imc_mail && GET_LEVEL(ch) < MIN_IMC_MAIL_LEVEL) {
!     send_to_char("You need to be at least level 15 to send notes to other muds.\r\n", ch);
!     return;
!   }
!   if (((recipient = get_id_by_name(buf)) < 0) && !imc_mail) {
      act("$n tells you, 'No one by that name is registered here!'",
  	FALSE, mailman, 0, ch, TO_VICT);
      return;
***************
*** 474,480 ****
    GET_GOLD(ch) -= STAMP_PRICE;
    SET_BIT(PLR_FLAGS(ch), PLR_MAILING | PLR_WRITING);
  
!   ch->desc->mail_to = recipient;
    ch->desc->str = (char **) malloc(sizeof(char *));
    *(ch->desc->str) = NULL;
    ch->desc->max_str = MAX_MAIL_SIZE;
--- 488,498 ----
    GET_GOLD(ch) -= STAMP_PRICE;
    SET_BIT(PLR_FLAGS(ch), PLR_MAILING | PLR_WRITING);
  
!   if (imc_mail) {
!     ch->desc->mail_to = -1;
!     ch->desc->imc_mail_to = addr;
!   } else
!     ch->desc->mail_to = recipient;
    ch->desc->str = (char **) malloc(sizeof(char *));
    *(ch->desc->str) = NULL;
    ch->desc->max_str = MAX_MAIL_SIZE;
