*** osrc/mail.h	Sat Apr 13 04:39:21 1996
--- src/mail.h	Sun Jun  8 23:48:25 1997
***************
*** 16,31 ****
  
  /* You can modify the following constants to fit your own MUD.  */
  
! /* minimum level a player must be to send mail	*/
  #define MIN_MAIL_LEVEL 2
  
! /* # of gold coins required to send mail	*/
  #define STAMP_PRICE 150
  
! /* Maximum size of mail in bytes (arbitrary)	*/
  #define MAX_MAIL_SIZE 4096
  
! /* size of mail file allocation blocks		*/
  #define BLOCK_SIZE 100
  
  /*
--- 16,34 ----
  
  /* You can modify the following constants to fit your own MUD.  */
  
! /* minimum level a player must be to send mail		*/
  #define MIN_MAIL_LEVEL 2
  
! /* minimum level a player must be to send IMC mail	*/
! #define MIN_IMC_MAIL_LEVEL 15
! 
! /* # of gold coins required to send mail		*/
  #define STAMP_PRICE 150
  
! /* Maximum size of mail in bytes (arbitrary)		*/
  #define MAX_MAIL_SIZE 4096
  
! /* size of mail file allocation blocks			*/
  #define BLOCK_SIZE 100
  
  /*
