*** osrc/Makefile.in	Sat Apr 13 19:46:37 1996
--- src/Makefile.in	Mon Jun  9 01:39:57 1997
***************
*** 21,27 ****
  	castle.o class.o config.o constants.o db.o fight.o graph.o handler.o \
  	house.o interpreter.o limits.o magic.o mail.o mobact.o modify.o \
  	objsave.o olc.o shop.o spec_assign.o spec_procs.o spell_parser.o \
! 	spells.o utils.o weather.o @LIBOBJS@
  
  default: .accepted
  	$(MAKE) ../bin/circle
--- 21,28 ----
  	castle.o class.o config.o constants.o db.o fight.o graph.o handler.o \
  	house.o interpreter.o limits.o magic.o mail.o mobact.o modify.o \
  	objsave.o olc.o shop.o spec_assign.o spec_procs.o spell_parser.o \
! 	spells.o utils.o weather.o imc-config.o imc-events.o imc-interp.o \
! 	imc-mail.o imc-util.o imc-version.o imc.o imc-mercbase.o @LIBOBJS@
  
  default: .accepted
  	$(MAKE) ../bin/circle
***************
*** 205,207 ****
--- 206,225 ----
  weather.o: weather.c conf.h sysdep.h structs.h utils.h comm.h handler.h \
    interpreter.h db.h
  	$(CC) -c $(CFLAGS) weather.c
+ imc-config.o: imc-config.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-config.c
+ imc-events.o: imc-events.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-events.c
+ imc-interp.o: imc-interp.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-interp.c
+ imc-mail.o: imc-mail.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-mail.c
+ imc-mercbase.o: imc-mercbase.c conf.h sysdep.h structs.h utils.h \
+   interpreter.h comm.h handler.h mail.h db.h imc.h imc-config.h imc-mercbase.h
+ 	$(CC) -c $(CFLAGS) imc-mercbase.c
+ imc-util.o: imc-util.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-util.c
+ imc-version.o: imc-version.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc-version.c
+ imc.o: imc.c imc.h imc-config.h
+ 	$(CC) -c $(CFLAGS) imc.c
