*** osrc/modify.c	Sat Apr 13 04:39:21 1996
--- src/modify.c	Mon Jun  9 01:21:48 1997
***************
*** 8,13 ****
--- 8,15 ----
  *  CircleMUD is based on DikuMUD, Copyright (C) 1990, 1991.               *
  ************************************************************************ */
  
+ #define CIRCLE
+ 
  #include "conf.h"
  #include "sysdep.h"
  
***************
*** 22,27 ****
--- 24,31 ----
  #include "mail.h"
  #include "boards.h"
  
+ #include "imc-mercbase.h"
+ 
  void show_string(struct descriptor_data *d, char *input);
  
  char *string_fields[] =
***************
*** 54,59 ****
--- 58,65 ----
  /* Add user input to the 'current' string (as defined by d->str) */
  void string_add(struct descriptor_data *d, char *str)
  {
+   char *tmstr;
+   time_t mytime;
    int terminator = 0;
    extern char *MENU;
  
***************
*** 90,99 ****
  
    if (terminator) {
      if (!d->connected && (PLR_FLAGGED(d->character, PLR_MAILING))) {
!       store_mail(d->mail_to, GET_IDNUM(d->character), *d->str);
        d->mail_to = 0;
        free(*d->str);
        free(d->str);
        SEND_TO_Q("Message sent!\r\n", d);
        if (!IS_NPC(d->character))
  	REMOVE_BIT(PLR_FLAGS(d->character), PLR_MAILING | PLR_WRITING);
--- 96,120 ----
  
    if (terminator) {
      if (!d->connected && (PLR_FLAGGED(d->character, PLR_MAILING))) {
!       if (d->mail_to != -1)
!         store_mail(d->mail_to, GET_IDNUM(d->character), *d->str);
!       else {
!         mytime = time(0);
!         tmstr = (char *) asctime(localtime(&mytime));
!         *(tmstr + strlen(tmstr) - 1) = '\0';
!         imc_post_mail(d->character,
!                       GET_NAME(d->character),
!                       d->imc_mail_to,
!                       tmstr,
!                       "MudMail",
!                       *d->str);
!       }
        d->mail_to = 0;
        free(*d->str);
        free(d->str);
+       if (d->imc_mail_to)
+         free(d->imc_mail_to);
+                         
        SEND_TO_Q("Message sent!\r\n", d);
        if (!IS_NPC(d->character))
  	REMOVE_BIT(PLR_FLAGS(d->character), PLR_MAILING | PLR_WRITING);
