*** osrc/structs.h	Sat Apr 13 04:39:22 1996
--- src/structs.h	Sun Jun  8 23:52:13 1997
***************
*** 740,748 ****
     int spare16;
     long	spare17;
     long	spare18;
!    long	spare19;
!    long	spare20;
!    long	spare21;
  };
  
  /*
--- 740,748 ----
     int spare16;
     long	spare17;
     long	spare18;
!    long imc_deaf;		/* IMC deaf-to_channel flags		*/
!    long imc_allow;		/* IMC channel allow overrides		*/
!    long imc_deny;		/* IMC channel deny overrides		*/
  };
  
  /*
***************
*** 761,766 ****
--- 761,769 ----
     long last_tell;		/* idnum of last tell from		*/
     void *last_olc_targ;		/* olc control				*/
     int last_olc_mode;		/* olc control				*/
+    
+    char *rreply;		/* IMC reply-to				*/
+    char *rreply_name;		/* IMC reply-to shown to char		*/
  };
  
  
***************
*** 888,893 ****
--- 891,897 ----
     char	**str;			/* for the modify-str system		*/
     size_t	max_str;		/*		-			*/
     long	mail_to;		/* name for mail system			*/
+    char *imc_mail_to;		/* name for IMC mail system		*/
     int	prompt_mode;		/* control of prompt-printing		*/
     char	inbuf[MAX_RAW_INPUT_LENGTH];  /* buffer for raw input		*/
     char	last_input[MAX_INPUT_LENGTH]; /* the last input			*/
