--- src.old/act_wiz.c	Fri Feb  7 03:22:47 1997
+++ src/act_wiz.c	Fri Apr 18 15:14:48 1997
@@ -4950,6 +4950,8 @@ void do_cset( CHAR_DATA *ch, char *argum
 	    sysdata.read_all_mail, sysdata.read_mail_free, sysdata.write_mail_free );
     ch_printf(ch, "  Take all mail: %d.\n\r",
 	    sysdata.take_others_mail);
+    ch_printf(ch, "  IMC board vnum: %d.\n\r",
+	    sysdata.imc_mail_vnum);
     ch_printf(ch, "Channels:\n\r  Muse: %d. Think: %d. Log: %d. Build: %d.\n\r",
  	    sysdata.muse_level, sysdata.think_level, sysdata.log_level, 
 	    sysdata.build_level);
@@ -5149,6 +5151,12 @@ void do_cset( CHAR_DATA *ch, char *argum
   {
     sysdata.level_mset_player = level;
     send_to_char("Ok.\n\r", ch);      
+    return;
+  }
+  if (!str_cmp(arg, "imc_mail_vnum"))
+  {
+    sysdata.imc_mail_vnum = level;
+    send_to_char("Ok.\n\r", ch);
     return;
   }
   else
