--- src.old/boards.c	Fri Feb  7 03:22:47 1997
+++ src/boards.c	Fri Apr 18 13:30:58 1997
@@ -21,6 +21,7 @@
 #include <time.h>
 #include <sys/stat.h>
 #include "mud.h"
+#include "imc-mercbase.h"
 
 /* Defines for voting on notes. -- Narn */
 #define VOTE_NONE 0
@@ -828,6 +829,8 @@ void do_note( CHAR_DATA *ch, char *arg_p
     {
 	struct stat fst;
 	char fname[1024];
+	bool imc = FALSE;
+	
 	if(get_trust(ch) < sysdata.write_mail_free )
 	{
 	    quill = find_quill( ch );
@@ -874,10 +877,20 @@ void do_note( CHAR_DATA *ch, char *arg_p
 
         arg_passed[0] = UPPER(arg_passed[0]);
 
+	if (strchr(arg_passed, '@')!=NULL)
+	{
+	    if (ch->level<15)
+	    {
+		send_to_char("You need to be at least level 15 to send notes to other muds.\n\r", ch);
+		return;
+	    }
+	    imc = TRUE;
+	}
+
         sprintf( fname, "%s%c/%s", PLAYER_DIR, tolower(arg_passed[0]),
                  capitalize( arg_passed ) );
  
-	if ( !IS_MAIL || stat( fname, &fst ) != -1 || !str_cmp(arg_passed, "all") )
+	if ( !IS_MAIL || imc || stat( fname, &fst ) != -1 || !str_cmp(arg_passed, "all") )
 	{                                       
 	    paper->value[2] = 1;
 	    ed = SetOExtra(paper, "_to_");
@@ -922,7 +935,8 @@ void do_note( CHAR_DATA *ch, char *arg_p
 
     if ( !str_cmp( arg, "post" ) )
     {
-	char *strtime, *text;
+	char *strtime, *to, *subj, *text;
+	bool imc = FALSE;
 
 	if ( ( paper = get_eq_char(ch, WEAR_HOLD) ) == NULL
 	||     paper->item_type != ITEM_PAPER )
@@ -949,11 +963,43 @@ void do_note( CHAR_DATA *ch, char *arg_p
 	    return;
 	}
 
+	strtime				= ctime( &current_time );
+	strtime[strlen(strtime)-1]	= '\0';
+
+	/* handle IMC notes */
+	to = get_extra_descr( "_to_", paper->first_extradesc );
+	subj = get_extra_descr( "_subject_", paper->first_extradesc );
+	text = get_extra_descr( "_text_", paper->first_extradesc );
+
+	if (to && strchr(to, '@')!=NULL)
+	{
+	  if ( !subj || !*subj )
+	  {
+	    send_to_char( "You must specify a subject for IMC mail.\n\r", ch );
+	    return;
+	  }
+	  if ( !text || !*text )
+	  {
+	    send_to_char( "You must have text in IMC mail.\n\r", ch );
+	    return;
+	  }
+
+	  {
+	    send_to_char( "The IMC board is not here.\n\r", ch );
+	    return;
+	  }
+	  imc = TRUE;
+	}
 	board = find_board( ch );
 	if ( !board )
 	{
-	    send_to_char( "There is no bulletin board here to post your note on.\n\r", ch );
-	    return;
+	  send_to_char( "There is no bulletin board here to post your note on.\n\r", ch );
+	  return;
+	}
+	if ( imc && board->board_obj != sysdata.imc_mail_vnum )
+	{
+	  send_to_char( "You can only post IMC mail on the IMC board.\n\r", ch );
+	  return;
 	}
 	if ( !can_post( ch, board ) ) 
 	{
@@ -969,22 +1015,21 @@ void do_note( CHAR_DATA *ch, char *arg_p
 
 	act( AT_ACTION, "$n posts a note.", ch, NULL, NULL, TO_ROOM );
 
-	strtime				= ctime( &current_time );
-	strtime[strlen(strtime)-1]	= '\0';
 	CREATE( pnote, NOTE_DATA, 1 );
 	pnote->date			= STRALLOC( strtime );
 
-	text = get_extra_descr( "_text_", paper->first_extradesc );
+	pnote->to_list = to ? STRALLOC( to ) : STRALLOC( "all" );
 	pnote->text = text ? STRALLOC( text ) : STRALLOC( "" );
-	text = get_extra_descr( "_to_", paper->first_extradesc );
-	pnote->to_list = text ? STRALLOC( text ) : STRALLOC( "all" );
-	text = get_extra_descr( "_subject_", paper->first_extradesc );
-	pnote->subject = text ? STRALLOC( text ) : STRALLOC( "" );
+	pnote->subject = subj ? STRALLOC( subj ) : STRALLOC( "" );
 	pnote->sender  = QUICKLINK( ch->name );
         pnote->voting      = 0;
         pnote->yesvotes    = str_dup( "" );
         pnote->novotes     = str_dup( "" );
         pnote->abstentions = str_dup( "" );
+        
+        if ( imc )
+          imc_post_mail(ch, pnote->sender, pnote->to_list, pnote->date,
+              pnote->subject, pnote->text);
 
 	LINK( pnote, board->first_note, board->last_note, next, prev );
 	board->num_posts++;
