--- src.old/db.c	Fri Feb  7 03:22:48 1997
+++ src/db.c	Fri Apr 18 12:09:17 1997
@@ -312,6 +312,7 @@ void boot_db( void )
     sysdata.read_mail_free 		= LEVEL_IMMORTAL;
     sysdata.write_mail_free 		= LEVEL_IMMORTAL;
     sysdata.take_others_mail		= LEVEL_DEMI;
+    sysdata.imc_mail_vnum		= 0;
     sysdata.muse_level			= LEVEL_DEMI;
     sysdata.think_level			= LEVEL_HIGOD;
     sysdata.build_level			= LEVEL_DEMI;
@@ -5366,6 +5367,7 @@ void save_sysdata( SYSTEM_DATA sys )
 	fprintf( fp, "Readmailfree   %d\n", sys.read_mail_free		);
 	fprintf( fp, "Writemailfree  %d\n", sys.write_mail_free		);
 	fprintf( fp, "Takeothersmail %d\n", sys.take_others_mail	);
+	fprintf( fp, "IMCMailVnum    %d\n", sys.imc_mail_vnum		);
 	fprintf( fp, "Muse           %d\n", sys.muse_level		);
 	fprintf( fp, "Think          %d\n", sys.think_level		);
 	fprintf( fp, "Build          %d\n", sys.build_level		);
@@ -5442,6 +5444,10 @@ void fread_sysdata( SYSTEM_DATA *sys, FI
 	case 'H':
 	    KEY( "Highplayers",	   sys->alltimemax,	  fread_number( fp ) );
 	    KEY( "Highplayertime", sys->time_of_max,      fread_string_nohash( fp ) );
+	    break;
+
+	case 'I':
+	    KEY( "IMCMailVnum",    sys->imc_mail_vnum,    fread_number( fp ) );
 	    break;
 
 	case 'L':
