--- src.old/interp.c	Fri Feb  7 03:41:37 1997
+++ src/interp.c	Wed Nov 12 22:08:32 1997
@@ -21,6 +21,7 @@
 #include <string.h>
 #include <time.h>
 #include "mud.h"
+#include "icec-mercbase.h"
 
 /*
  * Externals
@@ -490,7 +491,8 @@ void interpret( CHAR_DATA *ch, char *arg
     if ( !found )
     {
 	if ( !check_skill( ch, command, argument )
-	&&   !check_social( ch, command, argument ) )
+	&&   !check_social( ch, command, argument )
+	&&   !icec_command_hook( ch, command, argument) )
 	{
 	    EXIT_DATA *pexit;
 
