--- src.old/Makefile	Fri Feb  7 03:38:20 1997
+++ src/Makefile	Wed Nov 12 21:50:36 1997
@@ -3,7 +3,7 @@ PROF    =
 NOCRYPT =
 #Uncomment the next line if you want request support
 #DBUGFLG = -DREQUESTS
-C_FLAGS = -g3 -Wall $(PROF) $(NOCRYPT) $(DBUGFLG)
+C_FLAGS = -g3 -Wall $(PROF) $(NOCRYPT) $(DBUGFLG) -DSMAUG
 L_FLAGS = $(PROF)
 
 O_FILES = act_comm.o act_info.o act_move.o act_obj.o act_wiz.o boards.o \
@@ -11,14 +11,16 @@ O_FILES = act_comm.o act_info.o act_move
 	  handler.o hashstr.o ibuild.o id.o interp.o magic.o makeobjs.o \
 	  mapout.o misc.o mpxset.o mud_comm.o mud_prog.o player.o requests.o \
 	  reset.o save.o shops.o skills.o special.o tables.o track.o update.o \
-	  grub.o
+	  grub.o imc.o imc-mail.o imc-interp.o imc-util.o imc-config.o \
+	  imc-events.o imc-version.o imc-mercbase.o ice.o icec.o icec-mercbase.o
 
 C_FILES = act_comm.c act_info.c act_move.c act_obj.c act_wiz.c boards.c \
 	  build.c clans.c comm.c comments.c const.c db.c deity.c fight.c \
 	  handler.c hashstr.c ibuild.c id.c interp.c magic.c makeobjs.c \
 	  mapout.c misc.c mpxset.c mud_comm.c mud_prog.c player.c requests.c \
 	  reset.c save.c shops.c skills.c special.c tables.c track.c update.c \
-	  grub.c
+	  grub.o imc.c imc-mail.c imc-interp.c imc-util.c imc-config.c \
+	  imc-events.c imc-version.c imc-mercbase.c ice.c icec.c icec-mercbase.c
 
 H_FILES = mud.h bet.h
 
