--- src.old/mud.h	Fri Feb  7 03:22:50 1997
+++ src/mud.h	Wed Nov 12 22:01:04 1997
@@ -1936,6 +1936,13 @@ struct	pc_data
     char *		subprompt;	/* Substate prompt */
     sh_int		pagerlen;	/* For pager (NOT menus) */
     bool		openedtourney;
+
+    long	imc_deaf;    /* IMC channel def flags */
+    long	imc_allow;   /* IMC channel allow flags */
+    long	imc_deny;    /* IMC channel deny flags */
+    char *	rreply;      /* IMC reply-to */
+	char *      rreply_name; /* IMC reply-to shown to char */
+	char *ice_listen;
 };
 
 
@@ -2184,6 +2191,7 @@ struct	system_data
     sh_int	read_mail_free;		/* Read mail for free (was 51) */
     sh_int	write_mail_free;	/* Write mail for free(was 51) */
     sh_int	take_others_mail;	/* Take others mail (was 54)   */
+    sh_int	imc_mail_vnum;		/* Board vnum for IMC mail     */
     sh_int	muse_level;		/* Level of muse channel */
     sh_int	think_level;		/* Level of think channel LEVEL_HIGOD*/
     sh_int	build_level;		/* Level of build channel LEVEL_BUILD*/
