--- src.old/save.c	Fri Feb  7 03:22:49 1997
+++ src/save.c	Wed Nov 12 22:02:48 1997
@@ -288,6 +288,16 @@ void fwrite_char( CHAR_DATA *ch, FILE *f
       fprintf( fp, "Wimpy        %d\n",	ch->wimpy		);
     if ( ch->deaf )
       fprintf( fp, "Deaf         %d\n",	ch->deaf		);
+    if ( !IS_NPC(ch) )
+    {
+      if ( ch->pcdata->imc_deaf )
+        fprintf( fp, "IMC          %ld\n",   ch->pcdata->imc_deaf );
+      if ( ch->pcdata->imc_allow )
+        fprintf( fp, "IMCAllow     %ld\n",   ch->pcdata->imc_allow );
+      if ( ch->pcdata->imc_deny )
+	    fprintf( fp, "IMCDeny      %ld\n",   ch->pcdata->imc_deny );
+      fprintf(fp, "ICEListen %s~\n", ch->pcdata->ice_listen);
+    }
     if ( ch->resistant )
       fprintf( fp, "Resistant    %d\n",	ch->resistant		);
     if ( ch->immune )
@@ -797,6 +807,7 @@ bool load_char_obj( DESCRIPTOR_DATA *d, 
 	ch->pcdata->o_range_lo		= 0;
 	ch->pcdata->o_range_hi		= 0;
 	ch->pcdata->wizinvis		= 0;
+	ch->pcdata->ice_listen      = str_dup( "" );
     }
     else
     {
@@ -1145,6 +1156,10 @@ void fread_char( CHAR_DATA *ch, FILE *fp
 
 	case 'I':
 	    KEY( "IllegalPK",	ch->pcdata->illegal_pk,	fread_number( fp ) );
+	    KEY ( "IMC",	ch->pcdata->imc_deaf,	fread_number( fp ) );
+	    KEY ( "IMCAllow",	ch->pcdata->imc_allow,	fread_number( fp ) );
+		KEY ( "IMCDeny",	ch->pcdata->imc_deny,	fread_number( fp ) );
+		KEY ( "ICEListen",  ch->pcdata->ice_listen, fread_string( fp ) );
 	    KEY( "Immune",	ch->immune,		fread_number( fp ) );
 	    break;
 
