--- src.old/tables.c	Sat Mar  8 09:53:32 1997
+++ src/tables.c	Wed Nov 12 22:13:16 1997
@@ -18,6 +18,8 @@
 #include <stdio.h>
 #include <string.h>
 #include "mud.h"
+#include "imc-mercbase.h"
+#include "icec-mercbase.h"
 
 #if defined(KEY)
 #undef KEY
@@ -283,8 +285,13 @@ DO_FUN *skill_function( char *name )
 	if ( !str_cmp( name, "do_hset" ))		return do_hset;
 	break;
     case 'i':
+	if ( !str_cmp( name, "do_ichannels" ))  return do_ichannels;
+	if ( !str_cmp( name, "do_icommand" ))   return do_icommand;
 	if ( !str_cmp( name, "do_ide" ))		return do_ide;
 	if ( !str_cmp( name, "do_idea" ))		return do_idea;
+	if ( !str_cmp( name, "do_ilist" ))      return do_ilist;
+	if ( !str_cmp( name, "do_imc" ))		return do_imc;
+	if ( !str_cmp( name, "do_imclist" ))		return do_imclist;
 	if ( !str_cmp( name, "do_immortalize" ))	return do_immortalize;
 	if ( !str_cmp( name, "do_immtalk" ))		return do_immtalk;
 	if ( !str_cmp( name, "do_induct" ))		return do_induct;
@@ -293,6 +300,8 @@ DO_FUN *skill_function( char *name )
 	if ( !str_cmp( name, "do_instazone" ))		return do_instazone;
 	if ( !str_cmp( name, "do_inventory" ))		return do_inventory;
 	if ( !str_cmp( name, "do_invis" ))		return do_invis;
+	if ( !str_cmp( name, "do_isetup" ))     return do_isetup;
+	if ( !str_cmp( name, "do_istats" ))		return do_istats;
 	break;
     case 'k':
 	if ( !str_cmp( name, "do_kick" ))		return do_kick;
@@ -315,6 +324,7 @@ DO_FUN *skill_function( char *name )
 	if ( !str_cmp( name, "do_low_purge" ))		return do_low_purge;
 	break;
     case 'm':
+        if ( !str_cmp( name, "do_mailqueue" ))		return do_mailqueue;
 	if ( !str_cmp( name, "do_mailroom" ))		return do_mailroom;
 	if ( !str_cmp( name, "do_make" ))		return do_make;
 	if ( !str_cmp( name, "do_makeboard" ))		return do_makeboard;
@@ -440,7 +450,12 @@ DO_FUN *skill_function( char *name )
 	if ( !str_cmp( name, "do_rank" ))		return do_rank;
 	if ( !str_cmp( name, "do_rassign" ))		return do_rassign;
 	if ( !str_cmp( name, "do_rat" ))		return do_rat;
+	if ( !str_cmp( name, "do_rbeep" ))		return do_rbeep;
+	if ( !str_cmp( name, "do_rchannels" ))		return do_rchannels;
+	if ( !str_cmp( name, "do_rchanset" ))		return do_rchanset;
+	if ( !str_cmp( name, "do_rconnect" ))		return do_rconnect;
 	if ( !str_cmp( name, "do_rdelete" ))		return do_rdelete;
+	if ( !str_cmp( name, "do_rdisconnect" ))	return do_rdisconnect;
 	if ( !str_cmp( name, "do_reboo" ))		return do_reboo;
 	if ( !str_cmp( name, "do_reboot" ))		return do_reboot;
 	if ( !str_cmp( name, "do_recho" ))		return do_recho;
@@ -467,14 +482,24 @@ DO_FUN *skill_function( char *name )
 	if ( !str_cmp( name, "do_retran" ))		return do_retran;
 	if ( !str_cmp( name, "do_return" ))		return do_return;
 	if ( !str_cmp( name, "do_revert" ))		return do_revert;
+	if ( !str_cmp( name, "do_rignore" ))		return do_rignore;
+	if ( !str_cmp( name, "do_rinfo" ))		return do_rinfo;
 	if ( !str_cmp( name, "do_rip" ))		return do_rip;
 	if ( !str_cmp( name, "do_rlist" ))		return do_rlist;
 	if ( !str_cmp( name, "do_rmenu" ))		return do_rmenu;
 	if ( !str_cmp( name, "do_rpedit" ))		return do_rpedit;
+	if ( !str_cmp( name, "do_rping" ))      return do_rping;
 	if ( !str_cmp( name, "do_rpstat" ))		return do_rpstat;
+	if ( !str_cmp( name, "do_rquery" ))		return do_rquery;
+	if ( !str_cmp( name, "do_rreply" ))		return do_rreply;
 	if ( !str_cmp( name, "do_rreset" ))		return do_rreset;
 	if ( !str_cmp( name, "do_rset" ))		return do_rset;
+	if ( !str_cmp( name, "do_rsockets" ))		return do_rsockets;
 	if ( !str_cmp( name, "do_rstat" ))		return do_rstat;
+	if ( !str_cmp( name, "do_rtell" ))		return do_rtell;
+	if ( !str_cmp( name, "do_rwho" ))		return do_rwho;
+	if ( !str_cmp( name, "do_rwhois" ))		return do_rwhois;
+	if ( !str_cmp( name, "do_rfinger" ))		return do_rfinger;
 	break;
     case 's':
 	if ( !str_cmp( name, "do_sacrifice" ))		return do_sacrifice;
@@ -796,6 +821,8 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_hset )		return "do_hset";
     if ( skill == do_ide )		return "do_ide";
     if ( skill == do_idea )		return "do_idea";
+    if ( skill == do_imc )		return "do_imc";
+    if ( skill == do_imclist )		return "do_imclist";
     if ( skill == do_immortalize )	return "do_immortalize";
     if ( skill == do_immtalk )		return "do_immtalk";
     if ( skill == do_induct )		return "do_induct";
@@ -804,6 +831,7 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_instazone )	return "do_instazone";
     if ( skill == do_inventory )	return "do_inventory";
     if ( skill == do_invis )		return "do_invis";
+    if ( skill == do_istats )		return "do_istats";
     if ( skill == do_kick )		return "do_kick";
     if ( skill == do_kill )		return "do_kill";
     if ( skill == do_languages )	return "do_languages";
@@ -821,6 +849,7 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_lookmap )		return "do_lookmap";
     if ( skill == do_low_purge )	return "do_low_purge";
     if ( skill == do_mailroom )		return "do_mailroom";
+    if ( skill == do_mailqueue )	return "do_mailqueue";
     if ( skill == do_make )		return "do_make";
     if ( skill == do_makeboard )	return "do_makeboard";
     if ( skill == do_makeclan )		return "do_makeclan";
@@ -935,7 +964,12 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_rank )		return "do_rank";
     if ( skill == do_rassign )		return "do_rassign";
     if ( skill == do_rat )		return "do_rat";
+    if ( skill == do_rbeep )		return "do_rbeep";
+    if ( skill == do_rchannels )	return "do_rchannels";
+    if ( skill == do_rchanset )		return "do_rchanset";
+    if ( skill == do_rconnect )		return "do_rconnect";
     if ( skill == do_rdelete )		return "do_rdelete";
+    if ( skill == do_rdisconnect )	return "do_rdisconnect";
     if ( skill == do_reboo )		return "do_reboo";
     if ( skill == do_reboot )		return "do_reboot";
     if ( skill == do_recho )		return "do_recho";
@@ -962,14 +996,23 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_retran )		return "do_retran";
     if ( skill == do_return )		return "do_return";
     if ( skill == do_revert )		return "do_revert";
+    if ( skill == do_rignore )		return "do_rignore";
+    if ( skill == do_rinfo )		return "do_rinfo";
     if ( skill == do_rip )		return "do_rip";
     if ( skill == do_rlist )		return "do_rlist";
     if ( skill == do_rmenu )		return "do_rmenu";
     if ( skill == do_rpedit )		return "do_rpedit";
     if ( skill == do_rpstat )		return "do_rpstat";
+    if ( skill == do_rquery )		return "do_rquery";
+    if ( skill == do_rreply )		return "do_rreply";
     if ( skill == do_rreset )		return "do_rreset";
     if ( skill == do_rset )		return "do_rset";
+    if ( skill == do_rsockets )		return "do_rsockets";
     if ( skill == do_rstat )		return "do_rstat";
+    if ( skill == do_rtell )		return "do_rtell";
+    if ( skill == do_rwho )		return "do_rwho";
+    if ( skill == do_rwhois )		return "do_rwhois";
+    if ( skill == do_rfinger )		return "do_rfinger";
     if ( skill == do_sacrifice )	return "do_sacrifice";
     if ( skill == do_save )		return "do_save";
     if ( skill == do_savearea )		return "do_savearea";
@@ -1053,7 +1096,13 @@ char *skill_name( DO_FUN *skill )
     if ( skill == do_wizlock )		return "do_wizlock";
     if ( skill == do_yell )		return "do_yell";
     if ( skill == do_zap )		return "do_zap";
-    if ( skill == do_zones )		return "do_zones";
+	if ( skill == do_zones )		return "do_zones";
+	if ( skill == do_icommand )     return "do_icommand";
+	if ( skill == do_isetup )       return "do_isetup";
+	if ( skill == do_ichannels )    return "do_ichannels";
+	if ( skill == do_ilist )        return "do_ilist";
+    if ( skill == do_rping )        return "do_rping";
+	
     return "reserved";
 }
 
