--- src.old/act_comm.c	Sun Feb 16 08:17:43 1997
+++ src/act_comm.c	Sun Jun 22 00:45:40 1997
@@ -34,7 +34,8 @@
 #include <string.h>
 #include <time.h>
 #include "merc.h"
-
+#include "imc.h"
+#include "imc-mercbase.h"
 
 
 /*
@@ -354,6 +355,13 @@ void do_note( CHAR_DATA *ch, char *argum
 
     if ( !str_cmp( arg, "to" ) )
     {
+
+    if (strchr(argument, '@')!=NULL && ch->level<15)
+    {
+    	send_to_char("You need to be at least level 15 to send notes to other muds.\n\r", ch);
+    	return;
+    }
+    
 	note_attach( ch );
 	free_string( ch->pnote->to_list );
 	ch->pnote->to_list = str_dup( argument );
@@ -427,6 +435,14 @@ void do_note( CHAR_DATA *ch, char *argum
 	free_string( ch->pnote->date );
 	ch->pnote->date			= str_dup( strtime );
 	ch->pnote->date_stamp           = current_time;
+
+	/* handle IMC notes */
+	if (strchr(ch->pnote->to_list, '@')!=NULL)
+	  imc_post_mail(ch, ch->pnote->sender,
+			ch->pnote->to_list,
+			ch->pnote->date,
+			ch->pnote->subject,
+			ch->pnote->text);
 
 	if ( !note_list )
 	{
