--- src.old/comm.c	Sun Feb 16 08:17:43 1997
+++ src/comm.c	Wed Nov 12 21:04:32 1997
@@ -91,7 +91,8 @@ char version_str[] = "$VER: EnvyMud 2.0 
 #endif
 
 #include "merc.h"
-
+#include "imc.h"
+#include "icec.h"
 
 
 /*
@@ -457,9 +458,16 @@ int main( int argc, char **argv )
 #if defined( unix ) || defined( AmigaTCP ) || defined( WIN32 )
     control = init_socket( port );
     boot_db( );
+
+	imc_startup ("imc/"); /* start up IMC*/
+	icec_init();
+
     sprintf( log_buf, "EnvyMud is ready to rock on port %d.", port );
     log_string( log_buf );
     game_loop_unix( control );
+
+    imc_shutdown(); /* shut down IMC */
+
 #if !defined( WIN32 )
     close( control );
 #else
@@ -773,6 +781,9 @@ void game_loop_unix( int control )
 	    FD_SET( d->descriptor, &exc_set );
 	}
 
+	/* IMC */
+	maxdesc=imc_fill_fdsets(maxdesc, &in_set, &out_set, &exc_set);
+
 	if ( select( maxdesc+1, &in_set, &out_set, &exc_set, &null_time ) < 0 )
 	{
 	    perror( "Game_loop: select: poll" );
@@ -849,6 +860,9 @@ void game_loop_unix( int control )
 	    }
 	}
 
+
+	/* IMC */
+	imc_idle_select(&in_set, &out_set, &exc_set, current_time);
 
 
 	/*
