--- src.old/Makefile.std	Sun Feb 16 08:17:43 1997
+++ src/Makefile.std	Wed Nov 12 21:16:01 1997
@@ -6,12 +6,13 @@ PROF    = 
 DEBUG   = -g
 NOCRYPT =
 O_FLAGS = 
-C_FLAGS = $(O_FLAGS) -Wall $(DEBUG) $(PROF) $(NOCRYPT)
+C_FLAGS = $(O_FLAGS) -Wall $(DEBUG) $(PROF) $(NOCRYPT) -DENVY
 L_FLAGS = $(O_FLAGS)       $(DEBUG) $(PROF)
 
 O_FILES = act_comm.o act_info.o act_move.o act_obj.o act_wiz.o comm.o const.o \
           db.o fight.o handler.o interp.o magic.o save.o ssm.o special.o      \
-	  update.o
+	  update.o imc.o imc-interp.o imc-util.o imc-version.o imc-config.o   \
+	  imc-mail.o imc-events.o imc-mercbase.o ice.o icec.o icec-mercbase.o
 
 envy: $(O_FILES)
 	rm -f envy
