--- src.old/merc.h	Sun Feb 16 08:17:43 1997
+++ src/merc.h	Wed Nov 12 21:05:06 1997
@@ -972,6 +972,13 @@ struct	pc_data
     int                 pagelen;
     int 		learned		[ MAX_SKILL ];
     bool                switched;
+
+    long        imc_deaf;    /* IMC channel deaf flags */
+    long        imc_allow;   /* IMC channel allow flags */
+    long        imc_deny;    /* IMC channel deny flags */
+    char *      rreply;      /* IMC reply-to */
+	char *      rreply_name; /* IMC reply-to shown to char */
+	char *ice_listen;
 };
 
 /*
