--- src.old/save.c	Sun Feb 16 08:17:43 1997
+++ src/save.c	Wed Nov 12 21:06:47 1997
@@ -173,6 +173,10 @@ void fwrite_char( CHAR_DATA *ch, FILE *f
     fprintf( fp, "Armr        %d\n",	ch->armor		);
     fprintf( fp, "Wimp        %d\n",	ch->wimpy		);
     fprintf( fp, "Deaf        %d\n",	ch->deaf		);
+    fprintf( fp, "IMC         %ld\n",   ch->pcdata->imc_deaf    );
+    fprintf( fp, "IMCAllow    %ld\n",   ch->pcdata->imc_allow   );
+    fprintf( fp, "IMCDeny     %ld\n",   ch->pcdata->imc_deny    );
+    fprintf( fp, "ICEListen   %s~\n",   ch->pcdata->ice_listen  );
 
     if ( IS_NPC( ch ) )
     {
@@ -389,6 +393,8 @@ bool load_char_obj( DESCRIPTOR_DATA *d, 
 
     ch->pcdata->switched                = FALSE;
 
+    ch->pcdata->ice_listen      = str_dup( "" );
+	
     found = FALSE;
     fclose( fpReserve );
 
@@ -555,6 +561,14 @@ int fread_char( CHAR_DATA *ch, FILE *fp 
       { "Bmfout", TRUE,  DEFLT,			{ &ch->pcdata->bamfout,
 						                      NULL } },
       { "Immskll",TRUE,  DEFLT,			{ &ch->pcdata->immskll,
+						                      NULL } },
+      { "IMC",    FALSE, 0,                     { &ch->pcdata->imc_deaf,
+								      NULL } },
+      { "IMCAllow",FALSE,0,                     { &ch->pcdata->imc_allow,
+								      NULL } },
+      { "IMCDeny",FALSE, 0,                     { &ch->pcdata->imc_deny,
+								      NULL } },
+      { "ICEListen",TRUE,  DEFLT,			{ &ch->pcdata->ice_listen,
 						                      NULL } },
       { "Ttle",   TRUE,  (int) &def_title,	{ &ch->pcdata->title, NULL } },
       { "AtrPrm", FALSE, MAND,			{ &ch->pcdata->perm_str,
