--- src.old/comm.c	Thu Nov 25 11:09:41 1993
+++ src/comm.c	Wed Nov 12 21:32:50 1997
@@ -48,6 +48,8 @@
 #include <time.h>
 
 #include "merc.h"
+#include "imc.h"
+#include "icec.h"
 
 
 
@@ -148,8 +150,10 @@ int	socket		args( ( int domain, int type
 #endif
 
 #if	defined(linux)
+#if 0
 int	accept		args( ( int s, struct sockaddr *addr, int *addrlen ) );
 int	bind		args( ( int s, struct sockaddr *name, int namelen ) );
+#endif
 int	close		args( ( int fd ) );
 int	getpeername	args( ( int s, struct sockaddr *name, int *namelen ) );
 int	getsockname	args( ( int s, struct sockaddr *name, int *namelen ) );
@@ -396,9 +400,16 @@ int main( int argc, char **argv )
 #if defined(unix)
     control = init_socket( port );
     boot_db( );
+
+	imc_startup ("imc/");
+	icec_init();
+
     sprintf( log_buf, "Merc is ready to rock on port %d.", port );
     log_string( log_buf );
     game_loop_unix( control );
+
+    imc_shutdown(); /* shut down IMC */
+
     close( control );
 #endif
 
@@ -673,6 +684,9 @@ void game_loop_unix( int control )
 	    FD_SET( d->descriptor, &exc_set );
 	}
 
+	/* IMC */
+	maxdesc=imc_fill_fdsets(maxdesc, &in_set, &out_set, &exc_set);
+
 	if ( select( maxdesc+1, &in_set, &out_set, &exc_set, &null_time ) < 0 )
 	{
 	    perror( "Game_loop: select: poll" );
@@ -749,7 +763,10 @@ void game_loop_unix( int control )
 	    }
 	}
 
-
+	
+	/* IMC */
+	imc_idle_select(&in_set, &out_set, &exc_set, current_time);
+	
 
 	/*
 	 * Autonomous game motion.
@@ -777,7 +794,6 @@ void game_loop_unix( int control )
 		}
 	    }
 	}
-
 
 
 	/*
