--- src.old/interp.c	Thu Nov 25 13:34:13 1993
+++ src/interp.c	Wed Nov 12 21:48:02 1997
@@ -27,6 +27,8 @@
 #include <time.h>
 #include "merc.h"
 
+#include "imc-mercbase.h" /* contains declarations of do_xxx functions */
+#include "icec-mercbase.h"
 
 
 bool	check_social	args( ( CHAR_DATA *ch, char *command,
@@ -218,6 +220,19 @@ const	struct	cmd_type	cmd_table	[] =
 
 
 
+	/* IMC Mortal commands */
+    { "rtell",          do_rtell,       POS_SLEEPING,    0,  LOG_NORMAL },
+    { "rreply",         do_rreply,      POS_SLEEPING,    0,  LOG_NORMAL },
+    { "rwho",           do_rwho,        POS_SLEEPING,    5,  LOG_NORMAL },
+    { "rquery",         do_rquery,      POS_SLEEPING,    5,  LOG_NORMAL },
+    { "imc",            do_imc,         POS_DEAD,    L_GOD,  LOG_ALWAYS },
+    { "imclist",        do_imclist,     POS_DEAD,        5,  LOG_NORMAL },
+    { "rbeep",          do_rbeep,       POS_SLEEPING,    0,  LOG_NORMAL },
+    { "rfinger",        do_rfinger,     POS_SLEEPING,    5,  LOG_NORMAL },
+    { "rwhois",         do_rwhois,      POS_SLEEPING,    5,  LOG_NORMAL },
+    { "istats",         do_istats,      POS_SLEEPING,    0,  LOG_NORMAL },
+    { "rchannels",      do_rchannels,   POS_DEAD,        0,  LOG_NORMAL },
+
     /*
      * Immortal commands.
      */
@@ -298,6 +313,21 @@ const	struct	cmd_type	cmd_table	[] =
     { "mptransfer",     do_mptransfer,	POS_DEAD,       41,  LOG_NORMAL },
     { "mpforce",        do_mpforce,	POS_DEAD,       41,  LOG_NORMAL },
 
+
+    { "rinfo",		do_rinfo,	POS_DEAD,	 0,  LOG_NORMAL },
+    { "rsockets",	do_rsockets,	POS_DEAD,    L_ANG,  LOG_NORMAL },
+    { "rconnect",	do_rconnect,	POS_DEAD,    L_DEI,  LOG_ALWAYS },
+    { "rdisconnect",	do_rdisconnect,	POS_DEAD,    L_DEI,  LOG_ALWAYS },
+    { "rignore",	do_rignore,	POS_DEAD,    L_DEI,  LOG_ALWAYS },
+    { "mailq",		do_mailqueue,	POS_DEAD,    L_DEI,  LOG_NORMAL },
+    { "rchanset",	do_rchanset,	POS_DEAD,    L_DEI,  LOG_ALWAYS },
+	{ "rping",          do_rping,       POS_DEAD,    L_GOD,   LOG_NORMAL},
+
+	{ "icommand",       do_icommand,    POS_DEAD,    L_ANG,   LOG_NORMAL},
+	{ "isetup",         do_isetup,      POS_DEAD,    L_GOD,   LOG_ALWAYS},
+	{ "ilist",          do_ilist,       POS_DEAD,    L_ANG,   LOG_NORMAL},
+	{ "ichannels",      do_ichannels,   POS_DEAD,    L_ANG,   LOG_NORMAL},
+
     /*
      * End of list.
      */
@@ -1411,7 +1441,8 @@ void interpret( CHAR_DATA *ch, char *arg
 	 * Look for command in socials table.
 	 */
 	if ( !check_social( ch, command, argument ) )
-	    send_to_char( "Huh?\n\r", ch );
+        if (!icec_command_hook(ch, command, argument))
+			send_to_char( "Huh?\n\r", ch );
 	return;
     }
 
