--- src.old/comm.c	Thu Mar 20 16:56:28 1997
+++ src/comm.c	Mon Oct 27 17:42:18 1997
@@ -60,6 +60,12 @@
 #include "merc.h"
 #include "recycle.h"
 
+/* IMC headers */
+#include "imc.h"
+#include "imc-mercbase.h"
+#include "icec.h"
+#include "icec-mercbase.h"
+
 /* command procedures needed */
 DECLARE_DO_FUN(do_help		);
 DECLARE_DO_FUN(do_look		);
@@ -415,9 +421,12 @@ int main( int argc, char **argv )
 #if defined(unix)
     control = init_socket( port );
     boot_db( );
+    imc_startup("imc/");  /* IMC startup */
+    icec_init();          /* ICE client */
     sprintf( log_buf, "ROM is ready to rock on port %d.", port );
     log_string( log_buf );
     game_loop_unix( control );
+    imc_shutdown();
     close (control);
 #endif
 
@@ -695,6 +704,9 @@ void game_loop_unix( int control )
 	    FD_SET( d->descriptor, &exc_set );
 	}
 
+	/* IMC */
+	maxdesc=imc_fill_fdsets(maxdesc, &in_set, &out_set, &exc_set);
+
 	if ( select( maxdesc+1, &in_set, &out_set, &exc_set, &null_time ) < 0 )
 	{
 	    perror( "Game_loop: select: poll" );
@@ -774,6 +786,9 @@ void game_loop_unix( int control )
 	}
 
 
+	/* IMC */
+	imc_idle_select(&in_set, &out_set, &exc_set, current_time);
+
 
 	/*
 	 * Autonomous game motion.
@@ -2330,10 +2345,26 @@ void page_to_char( const char *txt, CHAR
 #if defined(macintosh)
 	send_to_char(txt,ch);
 #else
-    ch->desc->showstr_head = alloc_mem(strlen(txt) + 1);
-    strcpy(ch->desc->showstr_head,txt);
-    ch->desc->showstr_point = ch->desc->showstr_head;
-    show_string(ch->desc,"");
+    if (ch->desc->showstr_head &&
+	(strlen(txt)+strlen(ch->desc->showstr_head)+1) < 32000)
+    {
+      char *temp=alloc_mem(strlen(txt) + strlen(ch->desc->showstr_head) + 1);
+      strcpy(temp, ch->desc->showstr_head);
+      strcat(temp, txt);
+      ch->desc->showstr_point = temp + 
+	(ch->desc->showstr_point - ch->desc->showstr_head);
+      free_mem(ch->desc->showstr_head, strlen(ch->desc->showstr_head) + 1);
+      ch->desc->showstr_head=temp;
+    }
+    else
+    {
+      if (ch->desc->showstr_head)
+	free_mem(ch->desc->showstr_head, strlen(ch->desc->showstr_head)+1);
+      ch->desc->showstr_head = alloc_mem(strlen(txt) + 1);
+      strcpy(ch->desc->showstr_head,txt);
+      ch->desc->showstr_point = ch->desc->showstr_head;
+      show_string(ch->desc,"");
+    }
 #endif
 }
 
