--- src.old/const.c	Wed Jul 24 21:30:04 1996
+++ src/const.c	Mon Apr  7 22:38:57 1997
@@ -35,7 +35,7 @@
 #include "merc.h"
 #include "magic.h"
 #include "interp.h"
-
+#include "imc-mercbase.h"
 
 /* item type list */
 const struct item_type		item_table	[]	=
@@ -112,6 +112,8 @@ const   struct wiznet_type      wiznet_t
    {	"snoops",	WIZ_SNOOPS,	L2 },
    {	"switches",	WIZ_SWITCHES,	L2 },
    {	"secure",	WIZ_SECURE,	L1 },
+   {    "imc",          WIZ_IMC,        L4 }, /* IMC */
+   {    "imc-debug",    WIZ_IMCDEBUG,   L1 }, /* IMC */
    {	NULL,		0,		0  }
 };
 
