--- src.old/interp.c	Wed Jul 24 21:30:06 1996
+++ src/interp.c	Wed Nov 12 16:10:26 1997
@@ -37,7 +37,8 @@
 #include <time.h>
 #include "merc.h"
 #include "interp.h"
-
+#include "imc-mercbase.h"
+#include "icec-mercbase.h"
 
 bool	check_social	args( ( CHAR_DATA *ch, char *command,
 			    char *argument ) );
@@ -354,6 +355,39 @@ const	struct	cmd_type	cmd_table	[] =
     { "prefix",		do_prefix,	POS_DEAD,	IM,  LOG_NORMAL, 1 },
 
     /*
+     * IMC commands
+     */
+
+    { "rtell",          do_rtell,       POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "rreply",         do_rreply,      POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "rwho",           do_rwho,        POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "rwhois",         do_rwhois,      POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "rquery",         do_rquery,      POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "rfinger",        do_rfinger,     POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "imc",            do_imc,         POS_DEAD,      L1,   LOG_ALWAYS, 1 },
+    { "imclist",        do_imclist,     POS_DEAD,       0,   LOG_NORMAL, 1 },
+    { "rbeep",          do_rbeep,       POS_SLEEPING,   0,   LOG_NORMAL, 1 },
+    { "istats",         do_istats,      POS_DEAD,       0,   LOG_NORMAL, 1 },
+    { "rchannels",      do_rchannels,   POS_DEAD,       0,   LOG_NORMAL, 1 },
+
+    /* IMC imm commands */
+
+    { "rinfo",          do_rinfo,       POS_DEAD,       0,   LOG_NORMAL, 1 },
+    { "rsockets",       do_rsockets,    POS_DEAD,      IM,   LOG_NORMAL, 1 },
+    { "rconnect",       do_rconnect,    POS_DEAD,      L2,   LOG_ALWAYS, 1 },
+    { "rdisconnect",    do_rdisconnect, POS_DEAD,      L2,   LOG_ALWAYS, 1 },
+    { "rignore",        do_rignore,     POS_DEAD,      L2,   LOG_ALWAYS, 1 },
+    { "rchanset",       do_rchanset,    POS_DEAD,      L5,   LOG_ALWAYS, 1 },
+    { "mailqueue",      do_mailqueue,   POS_DEAD,      L4,   LOG_NORMAL, 1 },
+
+    { "icommand",       do_icommand,    POS_DEAD,      IM,   LOG_NORMAL, 1 },
+    { "isetup",         do_isetup,      POS_DEAD,      IM,   LOG_NORMAL, 1 },
+    { "ilist",          do_ilist,       POS_DEAD,       0,   LOG_NORMAL, 1 },
+    { "ichannels",      do_ichannels,   POS_DEAD,       0,   LOG_NORMAL, 1 },
+
+    { "rping",          do_rping,       POS_DEAD,      L2,   LOG_NORMAL, 1 },
+	
+    /*
      * End of list.
      */
     { "",		0,		POS_DEAD,	 0,  LOG_NORMAL, 0 }
@@ -459,7 +493,8 @@ void interpret( CHAR_DATA *ch, char *arg
 	 * Look for command in socials table.
 	 */
 	if ( !check_social( ch, command, argument ) )
-	    send_to_char( "Huh?\n\r", ch );
+	    if ( !icec_command_hook(ch, command, argument) )
+		send_to_char( "Huh?\n\r", ch );
 	return;
     }
 
