--- src.old/merc.h	Wed Jul 24 21:30:06 1996
+++ src/merc.h	Sat Aug  9 00:05:26 1997
@@ -1476,6 +1476,14 @@ struct	pc_data
     bool              	confirm_delete;
     char *		alias[MAX_ALIAS];
     char * 		alias_sub[MAX_ALIAS];
+
+    long                imc_deaf;    /* IMC deaf-to-channel flags */
+    long                imc_allow;   /* IMC channel allow overrides */
+    long                imc_deny;    /* IMC channel deny overrides */
+    char *              rreply;      /* IMC reply-to */
+    char *              rreply_name; /* IMC reply-to shown to char */
+
+    char *              ice_listen;  /* ICE listening-to-channels list */
 };
 
 /* Data for generating characters -- only used during generation */
