--- src.old/note.c	Wed Jul 24 21:30:06 1996
+++ src/note.c	Mon Apr  7 23:30:43 1997
@@ -40,6 +40,8 @@
 #include "recycle.h"
 #include "tables.h"
 
+#include "imc-mercbase.h"
+
 /* globals from db.c for load_notes */
 #if !defined(macintosh)
 extern  int     _filbuf         args( (FILE *) );
@@ -943,11 +945,26 @@ void parse_note( CHAR_DATA *ch, char *ar
 	    return;
 	}
 
+        if (strchr(ch->pnote->to_list, '@')!=NULL && ch->level<15)
+        {
+          send_to_char("You need to be at least level 15 to send notes to other muds.\n\r", ch);
+          return;
+        }
+
 	ch->pnote->next			= NULL;
 	strtime				= ctime( &current_time );
 	strtime[strlen(strtime)-1]	= '\0';
 	ch->pnote->date			= str_dup( strtime );
 	ch->pnote->date_stamp		= current_time;
+
+	/* handle IMC notes */
+
+	if (strchr(ch->pnote->to_list, '@')!=NULL)
+	  imc_post_mail(ch, ch->pnote->sender,
+			ch->pnote->to_list,
+			ch->pnote->date,
+			ch->pnote->subject,
+			ch->pnote->text);
 
 	append_note(ch->pnote);
 	ch->pnote = NULL;
