--- src.old/save.c	Wed Jul 24 21:30:07 1996
+++ src/save.c	Mon Oct 27 17:46:18 1997
@@ -216,6 +216,10 @@ void fwrite_char( CHAR_DATA *ch, FILE *f
     if (ch->affected_by != 0)
 	fprintf( fp, "AfBy %s\n",   print_flags(ch->affected_by));
     fprintf( fp, "Comm %s\n",       print_flags(ch->comm));
+    fprintf( fp, "IMC %s\n",        print_flags(ch->pcdata->imc_deaf));
+    fprintf( fp, "IMCAllow %s\n",   print_flags(ch->pcdata->imc_allow));
+    fprintf( fp, "IMCDeny %s\n",    print_flags(ch->pcdata->imc_deny));
+    fprintf( fp, "ICEListen %s~\n", ch->pcdata->ice_listen);
     if (ch->wiznet)
     	fprintf( fp, "Wizn %s\n",   print_flags(ch->wiznet));
     if (ch->invis_level)
@@ -568,6 +572,8 @@ bool load_char_obj( DESCRIPTOR_DATA *d, 
     ch->pcdata->condition[COND_FULL]	= 48;
     ch->pcdata->condition[COND_HUNGER]	= 48;
 
+    ch->pcdata->ice_listen              = str_dup( "" );
+
     found = FALSE;
     fclose( fpReserve );
     
@@ -983,6 +989,10 @@ void fread_char( CHAR_DATA *ch, FILE *fp
 
 	case 'I':
 	    KEY( "Id",		ch->id,			fread_number( fp ) );
+            KEY( "IMC",         ch->pcdata->imc_deaf,     fread_flag( fp ) );
+	    KEY( "IMCAllow",    ch->pcdata->imc_allow,    fread_flag( fp ) );
+	    KEY( "IMCDeny",     ch->pcdata->imc_deny,     fread_flag( fp ) );
+	    KEY( "ICEListen",   ch->pcdata->ice_listen, fread_string( fp ) );
 	    KEY( "InvisLevel",	ch->invis_level,	fread_number( fp ) );
 	    KEY( "Inco",	ch->incog_level,	fread_number( fp ) );
 	    KEY( "Invi",	ch->invis_level,	fread_number( fp ) );
